tests/test_trainer.py

```diff
@@ -0,0 +1,135 @@
+import os
+from functools import partial
+from tempfile import NamedTemporaryFile
+
+import pytest
+import torch
+from pynktrombonegym.spaces import ObservationSpaceNames as OSN
+from torch.optim import SGD
+from torch.utils.tensorboard import SummaryWriter
+
+from src.env.array_action import ARRAY_ORDER as AO_act
+from src.env.array_voc_state import VSON
+from src.env.make_env import make_env
+from src.models.dreamer import Dreamer
+from src.trainer import Trainer
+
+# from src.models.components.agent import Agent # AgentのPRがマージされたら追加
+from tests.models.abc.dummy_classes import DummyAgent as Agent
+from tests.models.abc.dummy_classes import DummyAgent as DA
+from tests.models.abc.dummy_classes import DummyController as DC
+from tests.models.abc.dummy_classes import DummyObservationDecoder as DOD
+from tests.models.abc.dummy_classes import DummyObservationEncoder as DOE
+from tests.models.abc.dummy_classes import DummyPrior as DP
+from tests.models.abc.dummy_classes import DummyTransition as DT
+from tests.models.abc.dummy_classes import DummyWorld as DW
+
+env = make_env(["data/sample_target_sounds"])
+
+hidden_shape = (16,)
+ctrl_hidden_shape = (16,)
+state_shape = (8,)
+action_shape = (len(AO_act),)
+
+obs_space = env.observation_space
+voc_stats_shape = obs_space[VSON.VOC_STATE].shape
+rnn_input_shape = (8,)
+
+gen_sound_shape = obs_space[OSN.GENERATED_SOUND_SPECTROGRAM].shape
+tgt_sound_shape = obs_space[OSN.TARGET_SOUND_SPECTROGRAM].shape
+obs_shape = (24,)
+
+obs_enc = DOE(state_shape, obs_shape)
+obs_dec = DOD(
+    voc_stats_shape,
+    gen_sound_shape,
+)
+prior = DP(state_shape)
+trans = DT(hidden_shape)
+ctrl = DC(action_shape, ctrl_hidden_shape)
+
+
+d_world = partial(DW)
+d_agent = partial(DA, action_shape=action_shape)
+
+world_opt = partial(SGD, lr=1e-3)
+ctrl_opt = partial(SGD, lr=1e-3)
+
+dreamer_args = (trans, prior, obs_enc, obs_dec, ctrl, d_world, d_agent, world_opt, ctrl_opt)
+del env
+
+
+def make_ckpt_path_and_tensorboard():
+    p_dir = NamedTemporaryFile().name
```

Comment:
一時ディレクトリは`TemporaryDirectory`を使用します。
```suggestion
    p_dir = TemporaryDirectory().name
```
https://docs.python.org/ja/3/library/tempfile.html#tempfile.TemporaryDirectory